lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1236 @@
+        availability_zones=dict(type='list', aliases=['zones', 'az']),
+        backup_retention_period=dict(type='int'),
+        ca_certificate_identifier=dict(),
+        character_set_name=dict(),
+        copy_tags_to_snapshot=dict(),
+        db_cluster_identifier=dict(aliases=['cluster_id']),
+        db_instance_class=dict(aliases=['class']),
+        db_instance_identifier=dict(required=True, aliases=['instance_id', 'id']),
+        db_name=dict(),
+        db_parameter_group_name=dict(),
+        db_security_groups=dict(),
+        db_snapshot_identifier=dict(),
+        db_subnet_group_name=dict(),
+        domain=dict(),
+        domain_iam_role_name=dict(),
+        enable_cloudwatch_logs_exports=dict(type='list'),
+        enable_iam_database_authentication=dict(type='bool'),
+        enable_performance_insights=dict(type='bool'),
+        engine=dict(choices=[
+            "aurora",
+            "aurora-mysql",
+            "aurora-postgresql",
+            "mariadb",
+            "mysql",
+            "oracle-ee",
+            "oracle-se",
+            "oracle-se1",
+            "oracle-se2",
+            "postgres",
+            "sqlserver-ee",
+            "sqlserver-ex",
+            "sqlserver-se",
+            "sqlserver-web",
+        ]),
+        engine_version=dict(),
+        final_db_snapshot_identifier=dict(aliases=['final_snapshot_identifier']),
+        force_failover=dict(type='bool'),
+        iops=dict(type='int'),
+        kms_key_id=dict(),
+        license_model=dict(choices=['license-included', 'bring-your-own-license', 'general-public-license']),
+        master_user_password=dict(aliases=['password'], no_log=True),
+        master_username=dict(aliases=['username']),
+        monitoring_interval=dict(),
+        monitoring_role_arn=dict(),
+        multi_az=dict(type='bool'),
+        new_db_instance_identifier=dict(aliases=['new_instance_id', 'new_id']),
+        option_group_name=dict(),
+        performance_insights_kms_key_id=dict(),
+        performance_insights_retention_period=dict(),
+        port=dict(type='int'),
+        preferred_backup_window=dict(aliases=['backup_window']),
+        preferred_maintenance_window=dict(aliases=['maintenance_window']),
+        processor_features=dict(type='dict'),
+        promotion_tier=dict(),
+        publically_accessible=dict(type='bool'),
+        restore_time=dict(),
+        s3_bucket_name=dict(),
+        s3_ingestion_role_arn=dict(),
+        s3_prefix=dict(),
+        skip_final_snapshot=dict(type='bool', default=False),
+        snapshot_identifier=dict(),
+        source_db_instance_identifier=dict(),
+        source_engine=dict(choices=['mysql']),
+        source_engine_version=dict(),
+        storage_encrypted=dict(type='bool'),
+        storage_type=dict(choices=['standard', 'gp2', 'io1']),
+        tags=dict(type='dict'),
+        tde_credential_arn=dict(),
+        tde_credential_password=dict(no_log=True),
```

Comment:
TDE is also only available for MSSQL and Oracle database engines according to https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.Encryption.html - do we want to enforce that, or does the API give a nice error?

Reply:
Hmmm, I'm not sure the storage_type works with aurora-postgresql. Seeing errors like: ``StorageTypeNotSupportedFault: An error occurred (StorageTypeNotSupported) when calling the CreateDBInstance operation: Invalid storage type: standard`` and ``StorageTypeNotSupportedFault: An error occurred (StorageTypeNotSupported) when calling the CreateDBInstance operation: Invalid storage type: gp2``

Reply:
Replying to myself, this is an aurora limitation and it's specified in the API docs: https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html

Maybe we can add in the docstring for ``storage_type`` that it is only for non-aurora instances.